### DISCUSSION TITLE
The ability to update the productVariant title inside the update operation, apart from just updating the title and price.

### DISCUSSION THREAD

**Discussion by mareenanazir** [Shopify/function-examples#446] [Feedback: Cart Transform API]
Hi, I'm trying to update the product variant title, one which is shown below the actual product title. 

This is my graphql query, 
```
query RunInput {
  cart {
    lines {
      id
      cost {
        amountPerQuantity {
          amount
          currencyCode
        }
      }
      variantPrice: attribute(key: "_variantPrice") {
        value
      }
      merchandise {
        __typename
        ... on ProductVariant {
          id
          title
        }
      }
    }
  }
}
```

and thats how I'm trying to update the price and variant title, but its not working. 
```
export function run(input) {
  const operations = input.cart.lines.reduce(
    /** @param {CartOperation[]} acc */
    (acc, cartLine) => {
      const updateOperation = optionallyBuildUpdateOperation(
        cartLine
      );

      if (updateOperation) {
        return [...acc, { update: updateOperation }];
      }

      return acc;
    },
    []
  );

  return operations.length > 0 ? { operations } : NO_CHANGES;
};

/**
 * @param {RunInput['cart']['lines'][number]} cartLine
 */
function optionallyBuildUpdateOperation(
  { id: cartLineId, merchandise, cost, variantPrice }
) {

  if (variantPrice?.value) {
    return {
      cartLineId,
      price: {
        adjustment: {
          fixedPricePerUnit: {
            amount: (Number(variantPrice.value)).toFixed(2),
          }
        }
      },
      merchandise: {
        title: 'hello',
      }
    };
  }
  return null;
}
```

Can you please help if this is even possible?

**Comment by aaackerman** [comment]
Hi @mareenanazir , currently our [UpdateOperation](https://shopify.dev/docs/api/functions/reference/cart-transform#example-update-operation) API supports updating the cart line title. Here's an example of the expected schema for the operation:

```json
{
  "operations": [
    {
      "update": {
        "cartLineId": "gid://shopify/CartLine/1",
        "title": "Customized Line Item",
        "image": {
          "url": "https://cdn.shopify.com/[...]/custom-image.png"
        },
        "price": {
          "adjustment": {
            "fixedPricePerUnit": {
              "amount": "100"
            }
          }
        }
      }
    }
  ]
}
```

To update the title, you'll need to nest it under the update key rather than within merchandise. However, this operation only updates the cart lines, not the merchandise values. You could use this to show the cart with the product title and then the cart line title below. Would that solve your needs or do you need a specific subtitle? 

Thanks for this question and feedback! 😄

**Reply by mareenanazir** [reply]
Hi @aaackerman, thankyou for responding. 
Actually no, this is not what we are trying to do, our usecase is actually that we need to update the variant title and not the main title. 
In this image, we want to update `size-a5-(14.8-x-21-cm)-orientation-vertical-paper-weight-300-gsm-110lb-uncoated-choose-your-option-design-option-2` this to a specific name. 
<img width="536" alt="Screenshot 2024-02-09 at 10 34 52 PM" src="https://github.com/Shopify/function-examples/assets/36929145/2ba97545-2056-439d-a87e-0df68ff6d08b">

So the solution that you proposed won't work for us. Is there anything else we can do to achieve this?

Thanks again and waiting for your feedback.

**Reply by aaackerman** [reply]
Thanks for clarifying your requirements! 

> Is there anything else we can do to achieve this?

We don't have a capability for subtitles for variants via cart transform at this time. I will log this feedback! 🙏

**Comment by allanemerson** [comment]
@aaackerman We have a similar need, added as feedback here: https://github.com/Shopify/function-examples/discussions/415

**Comment by KSrugpadusa** [comment]
I also need this!
